app/src/main/java/org/oppia/app/player/state/InteractionAdapter.kt

```diff
@@ -0,0 +1,142 @@
+      }
+      VIEW_TYPE_ITEM_SELECTION -> {
+        val inflater = LayoutInflater.from(parent.getContext())
+        val binding =
+          DataBindingUtil.inflate<ItemSelectionInteractionItemsBinding>(
+            inflater,
+            R.layout.item_selection_interaction_items,
+            parent,
+            false
+          )
+        ItemSelectionViewHolder(binding)
+      }
+      else -> throw IllegalArgumentException("Invalid view type")
+    }
+  }
+
+  override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
+    when (holder.itemViewType) {
+      VIEW_TYPE_MULTIPLE_CHOICE -> (holder as MultipleChoiceViewHolder).bind(
+        itemList!!.get(position),
+        position,
+        mSelectedItem
+      )
+      VIEW_TYPE_ITEM_SELECTION -> (holder as ItemSelectionViewHolder).bind(
+        itemList!!.get(position),
+        position,
+        mSelectedItem
+      )
+    }
+  }
+
+  // Determines the appropriate ViewType according to the interaction type.
+  override fun getItemViewType(position: Int): Int {
+    return if (interactionId.equals("ItemSelectionInput")) {
+      VIEW_TYPE_ITEM_SELECTION
+    } else {
+      VIEW_TYPE_MULTIPLE_CHOICE
+    }
+  }
+
+  override fun getItemCount(): Int {
+    return itemList!!.size
+  }
+
+  private inner class ItemSelectionViewHolder(val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(rawString: String, position: Int, selectedPosition: Int) {
+      binding.setVariable(BR.htmlContent, rawString)
+      binding.executePendingBindings();
+      val htmlResult: Spannable = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
+        rawString,
+        binding.root.tv_item_selection_contents
+      )
+      binding.root.tv_item_selection_contents.text = htmlResult
+
+      binding.root.checkbox_container.setOnClickListener {
+        if (binding.root.item_selection_checkbox.isChecked)
+          binding.root.item_selection_checkbox.setChecked(false)
+        else
+          binding.root.item_selection_checkbox.setChecked(true)
+        notifyDataSetChanged()
+      }
+    }
+  }
+
+  private inner class MultipleChoiceViewHolder(val binding: ViewDataBinding) : RecyclerView.ViewHolder(binding.root) {
+    internal fun bind(rawString: String, position: Int, selectedPosition: Int) {
+      binding.setVariable(BR.htmlContent, rawString)
+      binding.executePendingBindings();
+      val htmlResult: Spannable = htmlParserFactory.create(entityType, explorationId).parseOppiaHtml(
+        rawString,
+        binding.root.tv_multiple_choice_contents
+      )
+      binding.root.tv_multiple_choice_contents.text = htmlResult
+
+      if (selectedPosition == position)
+        binding.root.multiple_choice_radio_button.setChecked(true)
+      else
+        binding.root.multiple_choice_radio_button.setChecked(false)
+
+      binding.root.radio_container.setOnClickListener {
+        mSelectedItem = getAdapterPosition()
```

Comment:
You can actually remove this `mSelectedItem`, you don't need this.

Reply:
its been used to manage radio buttons state.